examples/lstm_stateful.py

```diff
@@ -0,0 +1,206 @@
+Depending on the parameters, either only the stateful LSTM converges,
+or both the stateful and stateless LSTM converge.
+'''
+from __future__ import print_function
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+# --------------------
+# EDITABLE PARAMETERS
+# --------------------
+
+# timesteps to use in the output averaging of the input
+# e.g. use 2 for 2-point average
+tsteps = 2
+
+# number of elements ahead that are used to make the LSTM prediction
+# when lahead >= tsteps, the stateless LSTM can converge
+# similar to the stateful LSTM
+lahead = 1
+
+# length of input
+input_len = 1000
+
+# training parameters
+batch_size = 1
+epochs = 10
+
+# ----------------------------
+# DO NOT EDIT UNDER THIS LINE
+# ----------------------------
+
+print("*" * 33)
+if lahead >= tsteps:
+    print("STATELESS LSTM WILL ALSO CONVERGE")
+else:
+    print("STATELESS LSTM WILL NOT  CONVERGE")
+print("*" * 33)
+
+# add a few points to account for the
+# nan-values that will be dropped in preprocessing
+to_drop = max(tsteps - 1, lahead - 1)
+input_len += to_drop
+
+
+def gen_uniform_amp(amp=1, xn=10000):
+    """Generates uniform random data between
+    -amp and +amp
+    and of length xn
+
+    Arguments:
+        amp: maximum/minimum range of uniform data
+        xn: length of series
+    """
+    data_input = np.random.uniform(-1 * amp, +1 * amp, xn)
+    data_input = pd.DataFrame(data_input)
+    return data_input
+
+
+np.random.seed(1986)
+
+print('Generating Data...')
+data_input = gen_uniform_amp(amp=0.1, xn=input_len)
+
+# set the target to be a N-point average of the input
+expected_output = data_input.rolling(window=tsteps, center=False).mean()
+
+# when lahead > 1, need to stride the input
+# https://docs.scipy.org/doc/numpy/reference/generated/numpy.repeat.html
+if lahead > 1:
+    data_input = np.repeat(data_input.values, repeats=lahead, axis=1)
+    data_input = pd.DataFrame(data_input)
+    for i, c in enumerate(data_input.columns):
+        data_input[c] = data_input[c].shift(i)
+
+# drop the nan
+expected_output = expected_output[to_drop:]
+data_input = data_input[to_drop:]
+
+print('Input shape:', data_input.shape)
+print('Output shape:', expected_output.shape)
+print('Input head: ')
+print(data_input.head())
+print('Output head: ')
+print(expected_output.head())
+print('Input tail: ')
+print(data_input.tail())
+print('Output tail: ')
+print(expected_output.tail())
+
+print('Plotting input and expected output')
+plt.plot(data_input[0][:10], '.')
+plt.plot(expected_output[0][:10], '-')
+plt.legend(['Input', 'Expected output'])
+plt.title('Input')
+plt.show()
+
+
+from keras.models import Sequential
+from keras.layers import Dense, LSTM
+
+
+def create_model(stateful: bool):
+    model = Sequential()
+    model.add(LSTM(20,
+              input_shape=(lahead, 1),
+              batch_size=batch_size,
+              return_sequences=False,
+              stateful=stateful,
+              activation='tanh'))
```

Comment:
You can remove `activation` and `return_sequences` since they take the default values.